.github/workflows/codecov_ats.yml

```
@@ -126,44 +125,65 @@ jobs:
                 "--ignore=tests/sentry/search/events",
                 "--ignore=tests/sentry/ingest/ingest_consumer/test_ingest_consumer_kafka.py",
                 "--ignore=tests/sentry/region_to_control/test_region_to_control_kafka.py"
-              ]' >> $GITHUB_OUTPUT
-              echo 'EOF' >> $GITHUB_OUTPUT
-              echo ATS_TESTS_TO_SKIP="'[]'" >> "$GITHUB_OUTPUT"
-              echo "::error ATS failed"
-              exit 1
+              ]' > .artifacts/codecov_ats_tests_to_skip.json
+              echo '[]' > .artifacts/codecov_ats_tests_to_run.json
+              # If we reached this point it means that ATS failed with some error
+              tee <<< '{"ats_success": false, "error": "exception_raised"}' "$GITHUB_STEP_SUMMARY" ".artifacts/codecov_ats_result.json"
             fi
           env:
             CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
             CODECOV_STATIC_TOKEN: ${{ secrets.CODECOV_STATIC_TOKEN }}
+        - uses: actions/upload-artifact@v3
+          with:
+            name: ats_tests_to_run.json
+            path: .artifacts/codecov_ats_tests_to_run.json
+            if-no-files-found: error
+        - uses: actions/upload-artifact@v3
+          with:
+            name: ats_tests_to_skip.json
+            path: .artifacts/codecov_ats_tests_to_skip.json
+            if-no-files-found: error
+        - uses: actions/upload-artifact@v3
+          with:
+            name: ats_result.json
+            path: .artifacts/codecov_ats_result.json
+            if-no-files-found: error
```

Several files with the same prefix implies that a directory might be a good idea. It also means you can upload and download them as a group.

* .artifacts/codecov_ats/
  * tests_to_run.json
  * tests_to_skip.json
  * result.json

```suggestion
        - uses: actions/upload-artifact@v3
          with:
            name: codecov_ats
            path: .artifacts/codecov_ats
            if-no-files-found: error
```

P3: nit